internal/services/mobilenetwork/mobile_network_packet_core_control_plane_resource.go

```diff
@@ -0,0 +1,687 @@
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r PacketCoreControlPlaneResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.PacketCoreControlPlaneClient
+
+			id, err := packetcorecontrolplane.ParsePacketCoreControlPlaneID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := PacketCoreControlPlaneModel{
+				Name:              id.PacketCoreControlPlaneName,
+				ResourceGroupName: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+
+				if model.Properties.UeMtu != nil {
+					state.UeMtu = *model.Properties.UeMtu
+				}
+
+				state.Identity, err = flattenMobileNetworkUserAssignedToNetworkLegacyIdentity(model.Identity)
+				if err != nil {
+					return fmt.Errorf("flattening `identity`: %+v", err)
+				}
+
+				properties := model.Properties
+				state.ControlPlaneAccessInterface = flattenPacketCoreControlPlaneInterfacePropertiesModel(properties.ControlPlaneAccessInterface)
+
+				if properties.CoreNetworkTechnology != nil {
+					state.CoreNetworkTechnology = string(*properties.CoreNetworkTechnology)
+				}
+
+				if properties.InteropSettings != nil && *properties.InteropSettings != nil {
+					interopSettingsValue, err := json.Marshal(*properties.InteropSettings)
+					if err != nil {
+						return err
+					}
+
+					state.InteropSettings = string(interopSettingsValue)
+				}
+
+				state.LocalDiagnosticsAccess = flattenLocalPacketCoreControlLocalDiagnosticsAccessConfiguration(properties.LocalDiagnosticsAccess)
+
+				state.SiteIds = flattenPacketCoreControlPlaneSites(properties.Sites)
+
+				state.Platform = flattenPlatformConfigurationModel(properties.Platform)
+
+				state.Sku = string(properties.Sku)
+
+				if properties.Version != nil {
+					state.Version = *properties.Version
+				}
+				if model.Tags != nil {
+					state.Tags = *model.Tags
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r PacketCoreControlPlaneResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.PacketCoreControlPlaneClient
+
+			id, err := packetcorecontrolplane.ParsePacketCoreControlPlaneID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			if err := resourceMobileNetworkChildWaitForDeletion(ctx, id.ID(), func() (*http.Response, error) {
```

Comment:
is there a Swagger issue tracking this API bug? can we link it here in a comment as we do in every other resource that's blocked by a Swagger issue

Reply:
there is a comment on the definition of `resourceMobileNetworkChildWaitForDeletion`.